module/user-extension/helper.js

```diff
@@ -19,670 +18,785 @@ const programsHelper = require(MODULES_BASE_PATH + "/programs/helper");
+                platformRoles: {
+                  $elemMatch: { code: role, programs: ObjectId(programId) },
+                },
+              },
+            ],
+          },
+          {
+            _id: 1,
+          }
+        );
+
+        if (!userInformation) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: constants.apiResponses.USER_PLATFORM_ROLE_NOT_FOUND,
+          });
+        }
+
+        const programDocuments = await programsHelper.programDocuments(
+          {
+            _id: programId,
+            status: constants.common.ACTIVE,
+          },
+          ["components"]
+        );
+
+        if (!(programDocuments.length > 0)) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: constants.apiResponses.PROGRAM_NOT_FOUND,
+          });
+        }
+
+        const solutionDocuments = await solutionsHelper.solutionDocuments(
+          {
+            _id: { $in: programDocuments[0].components },
+            programId: programId,
+            isReusable: false,
+            isDeleted: false,
+          },
+          [
+            "externalId",
+            "description",
+            "name",
+            "type",
+            "subType",
+            "isRubricDriven",
+            "scoringSystem",
+            "criteriaLevelReport",
+          ]
+        );
+
+        return resolve({
+          message: constants.apiResponses.SOLUTIONS_LIST,
+          data: solutionDocuments,
+        });
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+
+  /**
+   * update the userExtension document.
+   * @method
+   * @name updateOne
+   * @param {Object} query    - findMatchQuery.
+   * @param {Object} setQuery - setQuery.
+   * @returns {Array}
+   */
 
-                const solutionDocuments = await solutionsHelper.solutionDocuments({
-                    _id: {$in: programDocuments[0].components},
-                    programId: programId,
-                    isReusable: false,
-                    isDeleted: false
-                },[
-                    "externalId",
-                    "description",
-                    "name",
-                    "type",
-                    "subType",
-                    "isRubricDriven",
-                    "scoringSystem",
-                    "criteriaLevelReport"
-                ]);
-       
-                return resolve({
-                   message: constants.apiResponses.SOLUTIONS_LIST,
-                   data: solutionDocuments
-                });
-            } catch(error) {
-                return reject(error);
-            }
-        })
-    }
+  static updateOne(query, setQuery, arrayFilters = []) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let updatedData = await database.models.userExtension.updateOne(
+          query,
+          setQuery,
+          { arrayFilters, upsert: true }
```

Comment:
@praveenKDass Make this similar to updateMany of the other modules. Change this function to do updateMany